Broken AppVeyor build

The current [Appveyor build](https://ci.appveyor.com/project/massich/pydicom/builds/20086930/job/w2b69xe3muieptq9) for Python 3.6 fails consistently with this message:
```
_____________ TestBadValueRead.test_read_bad_value_in_VR_default ______________
self = <pydicom.tests.test_valuerep.TestBadValueRead object at 0x000000BADBDC6EF0>
    def test_read_bad_value_in_VR_default(self):
        # found a conversion
        assert '1A' == convert_value('SH', self.tag)
        # converted with fallback vr "SH"
>       assert '1A' == convert_value('IS', self.tag)
E       AssertionError: assert '1A' == b'1A'
E        +  where b'1A' = convert_value('IS', <pydicom.tests.test_valuerep.TestBadValueRead.setup.<locals>.TagLike object at 0x000000BADBDB0588>)
E        +    where <pydicom.tests.test_valuerep.TestBadValueRead.setup.<locals>.TagLike object at 0x000000BADBDB0588> = <pydicom.tests.test_valuerep.TestBadValueRead object at 0x000000BADBDC6EF0>.tag
```

I couldn't reproduce this locally yet, using the same Python and pytest versions:
```
platform win32 -- Python 3.6.5, pytest-3.10.0, py-1.7.0, pluggy-0.8.0
rootdir: C:\dev\GitHub\pydicom, inifile:collected 1 item

test_valuerep.py .Expected length to be even multiple of number size
Expected length to be even multiple of number size
                                                       [100%]

========================== 1 passed in 0.27 seconds ===========================
Process finished with exit code 0
```

--- a/pydicom/tests/test_valuerep.py
+++ b/pydicom/tests/test_valuerep.py
@@ -196,6 +196,10 @@ class TestBadValueRead(object):
         self.tag.is_little_endian = True
         self.tag.is_implicit_VR = False
         self.tag.tag = Tag(0x0010, 0x0020)
+        self.default_retry_order = pydicom.values.convert_retry_VR_order
+
+    def teardown(self):
+        pydicom.values.convert_retry_VR_order = self.default_retry_order
 
     def test_read_bad_value_in_VR_default(self):
         # found a conversion

